### PR TITLE
Replace Legacy Request Data in Registrations Admin

### PR DESCRIPTION
**Pull request by tn3rb**
This PR simply updates the Registrations Admin by replacing all uses of the legacy `_req_data` array with functionality on the new Request class

### CODE REVIEW

**Review by Pebblo — APPROVED** [review]
Noice 👍

**Review by knazart — APPROVED** [review]
Like the changes. Looks good to me.

**Comment by Pebblo** [comment]
I spotted something whilst testing this although so far it hasn't caused issues as such.

The `_wp_http_referer` query string value has the slashes removed:

`&_wp_http_referer=wp-adminadmin.phppageespresso_registrations`

Prior to this branch it shows:

`&_wp_http_referer=%2Fwp-admin%2Fadmin.php%3Fpage%3Despresso_registrations`

To reproduce use the above branch.

Go to Event Espresso -> Registrations.

Select a month to filter down to (any will do, just select a month and click filter).

Now click on the the 'All Registrations CSV Report'.

Once you are redirected back to the registration list table look in the address bar.

**Comment by tn3rb** [comment]
good catch @Pebblo I'll look into that right now

**Comment by tn3rb** [comment]
the exception stack trace display class wasn't working so I fixed that too because i wanted to use that while debugging

File: core/exceptions/ExceptionStackTraceDisplay.php [hunk 1]
```diff
@@ -41,191 +53,185 @@ public function __construct(Exception $exception)
...
-        if (! WP_DEBUG) {
-            $output .= '
-	<p>';
+        $output = '
+<div id="ee-error-message" class="error">
+    ' . $error_message . '
+</div>';
+        $styles = $this->exceptionStyles();
+        $scripts = $this->printScripts(true);
+        if (defined('DOING_AJAX')) {
```
Comment (knazart) [inline core/exceptions/ExceptionStackTraceDisplay.php:76]: Should we also check for the value of this constant ?

File: core/exceptions/ExceptionStackTraceDisplay.php [hunk 2]
```diff
@@ -41,191 +53,185 @@ public function __construct(Exception $exception)
...
+        echo $styles, $output, $scripts; // already escaped
+    }
+
+
+    private function genericError($msg, $code)
+    {
+        return '
+    <p>
+        <span class="ee-error-user-msg-spn">' . trim($msg) . '</span> &nbsp; <sup>' . $code . '</sup>
+	</p>';
```
Comment (knazart) [inline core/exceptions/ExceptionStackTraceDisplay.php:89]: Kinda messed up spacing.

**Review by knazart — APPROVED** [review]
👍

This new `getRequestParam()` is a really nice addition.

File: core/exceptions/ExceptionStackTraceDisplay.php (hunk 1 above)
Comment (tn3rb) [reply core/exceptions/ExceptionStackTraceDisplay.php:76]: it is only ever set to `true` by WP core, so unless a third party is setting that constant to `false`,  it's value is kinda moot... 
ie: it's either `true` or undefined

File: core/exceptions/ExceptionStackTraceDisplay.php (hunk 2 above)
Comment (tn3rb) [reply core/exceptions/ExceptionStackTraceDisplay.php:89]: good eye, will fix, thnx

File: core/exceptions/ExceptionStackTraceDisplay.php (hunk 1 above)
Comment (knazart) [reply core/exceptions/ExceptionStackTraceDisplay.php:76]: Right, ok.

**Comment by Pebblo** [comment]
Tested various sections on the Event Espresso -> Registrations section.

Actions links.
Table views.
Table filters.
Exporting all registrations.
Filtered exports.
Event Check-in tab (selecting an event and checking in/out a registration)
Trashing and Perm Deleting registrations.

Then tested changing the registration status for a single registration and also retriggering messages for it.

Also tested the original reason for this change and that seems fixed.

@tn3rb I think the above covers this, right?

**Pull request merged**